compiler/vm/compilerbridge.nim

```diff
@@ -148,50 +161,115 @@ proc unpackResult(res: sink ExecutionResult; config: ConfigRef, node: PNode): PN
   else:
     let
       err = res.takeErr
-      errKind = err.report.kind
-      stId = config.addReport(wrap(
-        VMReport(
-          kind: rvmStackTrace,
-          currentExceptionA: err.stackTrace.currentExceptionA,
-          currentExceptionB: err.stackTrace.currentExceptionB,
-          traceReason: vmEventToLegacyReportKind(err.stackTrace.traceReason),
-          stacktrace: err.stackTrace.stacktrace,
-          skipped: err.stackTrace.skipped,
-          location: some err.location,
-          reportInst: toReportLineInfo(err.instLoc)
-          )))
-      rId = config.addReport(wrap(err.report))
-
-    result = config.newError(node, errKind, rId, instLoc(-1)):
-                             [newIntNode(nkIntLit, ord(stId))]
-
-proc buildError(c: TCtx, thread: VmThread, event: sink VMReport): ExecErrorReport  =
+      errKind = err.kind
+      astDiagTrace = AstDiagVmTrace(
+        currentExceptionA: err.stackTrace.currentExceptionA,
+        currentExceptionB: err.stackTrace.currentExceptionB,
+        traceReason: err.stackTrace.traceReason,
+        stacktrace: err.stackTrace.stacktrace,
+        skipped: err.stackTrace.skipped,
+        location: err.location,
+        instLoc: err.instLoc)
+    
+    result = config.newError(node):
+      case errKind
+      of execErrorVm:
+        let location =
+          case err.vmErr.kind
+          of vmEvtUserError:
+            err.vmErr.errLoc
+          of vmEvtArgNodeNotASymbol:
+            err.vmErr.argAst.info
+          else:
+            err.location
+
+        PAstDiag(
+          kind: adVmError,
+          location: location,
+          instLoc: err.vmErr.instLoc,
+          vmErr: vmEventToAstDiagVmError(err.vmErr),
+          vmTrace: astDiagTrace)
+      of execErrorVmGen:
+        PAstDiag(
+          kind: adVmGenError,
+          location: err.genErr.location,
+          instLoc: err.genErr.instLoc,
+          vmGenErr: vmGenDiagToAstDiagVmGenError(err.genErr),
+          duringJit: true,
+          vmGenTrace: astDiagTrace)
+      of execErrorQuit:
+        PAstDiag(
+          kind: adVmQuit,
+          location: err.location,
+          instLoc: err.instLoc,
+          vmExitCode: err.exitCode,
+          vmExitTrace: astDiagTrace)
```

Comment:
This is all a bit bloated, but I'm pretty sure I can ditch `vmlegacy` by the end of this.